### ISSUE TITLE
Issues with auto-exposure-mode

### ISSUE THREAD

**Issue by nickrobillard** [jtfrey/uvc-util#6]
Hi. This may be a support request or may be a bug.

I'm using macOS Catalina 10.15.5 (19F101) with Xcode Version 11.5 (11E608c). I am using a USB 2 Angetube webcam.

All commands I have tried so far are working as expected - with the exception of `auto-exposure-mode`. Note that, using a demo version of `iGlasses`, I have been able to change exposure between auto and manual, and then change the manual value successfully, so I know that my camera supports this activity in some form.

1. Here is what I have tried with `uvc-util`. First list usb camera devices:
```
$ ./uvc-util -d
------------ -------------- ------------ ------------ ------------------------------------------------
Index        Vend:Prod      LocationID   UVC Version  Device name
------------ -------------- ------------ ------------ ------------------------------------------------
0            0x0c45:0x6536  0x14600000   1.00         USB  Live camera
------------ -------------- ------------ ------------ ------------------------------------------------
```

2. List controls for index 0:
```
$ ./uvc-util -I 0 -c
UVC controls implemented by this device:
  brightness
  white-balance-temp
  roll-rel
  backlight-compensation
  zoom-abs
  pan-tilt-abs
  exposure-time-abs
  auto-focus
  power-line-frequency
  exposure-time-rel
  auto-exposure-priority
  gain
  saturation
  roll-abs
  auto-exposure-mode
  sharpness
  contrast
  hue
  gamma
  focus-rel
  auto-white-balance-temp
  focus-abs
```

3. Show control info for `auto-exposure-mode`:
```
$ ./uvc-util -I 0 -S auto-exposure-mode
auto-exposure-mode {
  type-description: {
    single value, unsigned 8-bit integer
  },
  step-size: 9
  default-value: 8
}
```

4. Get control value for `auto-exposure-mode` also shows current value of 8:
```
$ ./uvc-util -I 0 -g auto-exposure-mode
auto-exposure-mode = 8
```

5. My camera is currently in auto exposure mode (aka: continuous) so I assume this correlates with auto exposure mode being enabled. However, I am unable to change the `auto-exposure-mode` to `manual`. I have tried integers from 0 to 20 just to see what the effect is and I always get this:
```
$ ./uvc-util -I 0 -s auto-exposure-mode=0
ERROR:  unable to write new value to control auto-exposure-mode
```

6. However, the following does work as expected. I am able to put the camera back into `auto` exposure mode after being set to `manual` mode in `iGlasses`:
```
$ ./uvc-util -I 0 -s auto-exposure-mode=default
```
I observe in `iGlasses` that auto exposure mode toggle is immediately switched to `auto` as expected. So setting back to the default of `auto` exposure works. However, this only works when using the string value `default`. I would expect integer `8` to also work, but it shows the same error output as step 5 above.

7. The following also works as expected. After setting the exposure mode to `manual` using `iGlasses` app, I am able to change the exposure value using `uvc-util` with the following:
```
$ ./uvc-util -I 0 -s exposure-time-abs=300
```
And I immediately see the camera's exposure changed wherever I am viewing it.

Thinking that maybe `iGlasses` app is interfering somehow with `uvc-util`, I have also closed the `iGlasses` app and restarted my system (in attempts to clear any locks) but the same behavior from step 5 persists.

I hope this is user error on my part. I am happy to provide any more info that may be helpful. Thank you very much for your time and effort with this tool.

**Comment by jtfrey** [comment]
It's quite possible Apple has changed the underlying frameworks/libraries in ways that make this code incompatible with newer releases of the OS.  The program was written for another IT unit I was helping at the time, so I am not maintaining this program moving forward.  I encourage other developer(s) to take over.

**Comment by dhughes-xumak** [comment]
Fixed by #9.
